### ISSUE TITLE
Intellisense Problem: exception specification for virtual function

### ISSUE THREAD

**Issue by lh123** [microsoft/vscode-cpptools#1942]
* Operating System: Win10 64bit
 * VS Code version: 1.23.0
 * C/C++ extension version: 0.16.1
 * C/C++ Compiler: MinGW-W64

**Base class:**
![image](https://user-images.githubusercontent.com/10697249/39670162-2900781a-5131-11e8-84ae-9da446cfbb09.png)

**Subclass:**
![image](https://user-images.githubusercontent.com/10697249/39670156-1cb4582e-5131-11e8-9477-2b1010c67fdd.png)

**Comment by sean-mcmanus** [comment]
I'm not able repro this. Did you set your compilerPath to the mingw-64 gcc.exe? What is your cppStandard? Does it repro if you put the UIComponent and GameView in the same file?

**Comment by lh123** [comment]
@sean-mcmanus 

**I also found this error just exists in this class**

![image](https://user-images.githubusercontent.com/10697249/39694026-763f0e4c-5218-11e8-96a5-6f1de7f45aa1.png)

```
 "cStandard": "c11",
 "cppStandard": "c++14"
```

* When I put `UIComponent`  and  `GameView` in the same file, the error still exists

* When I add `noexcept` to this function, the error disappears.

![image](https://user-images.githubusercontent.com/10697249/39693208-da03b7c8-5215-11e8-9a56-7e489cd14cd8.png)

here is my code

[GameView.h](https://github.com/lh123/NesEmulator/blob/3c83e3e5774bede91b78a0f75e7c6741add8f6f8/include/ui/GameView.h)
[UIComponent.h](https://github.com/lh123/NesEmulator/blob/3c83e3e5774bede91b78a0f75e7c6741add8f6f8/include/ui/UIComponent.h)

**Comment by sean-mcmanus** [comment]
Okay, I'm able to repro it now -- strangely, the `#include <mutex>` along with "std::mutex mFrameMutex;"  is required for the repro. The bug doesn't repro with "msvc-x64" mode (i.e. non-gcc compiler). Minimum repro is below:
```
class UIComponent {
public:
    virtual ~UIComponent();
};

class foo
{
    foo() noexcept {};
};

class GameView : public UIComponent {
public:
    virtual ~GameView();

private:
    foo mFrameMutex;
};
```

**Comment by sean-mcmanus** [comment]
A bug has been assigned to the VS C++ team responsible. Thanks for reporting this. No ETA on a fix yet.

UPDATE: FYI, the fix should be in our 0.21.0-insiders release next week.

**Comment by sean-mcmanus** [comment]
Should be fixed with 0.21.0-insiders (auto installed if `"C_Cpp.updateChannel": "Insiders"` is set, or the [offline vsix](https://github.com/Microsoft/vscode-cpptools/releases/tag/0.21.0-insiders)). Let us know if you find any problems with it.
